bridge/contracts/BToken.sol

```diff
@@ -82,6 +84,35 @@ contract BToken is
         _setSplitsInternal(332, 332, 332, 4);
     }
 
+    /// @dev
+    ///
+    function payAndDeposit(
+        uint256 maxEth,
+        uint256 maxTokens,
+        bytes calldata data
+    ) public payable {
+        //forward call to btoken
+        uint256 bTokenFee = BridgeRouter(_bridgeRouterAddress()).routeDeposit(
+            msg.sender,
+            maxTokens,
+            data
+        );
+        //if the message has value (eth payment) require the value of eth equal btokenAmount, else destroy btoken amount specified
+        if (msg.value > 0) {
+            uint256 ethFee = _bTokensToEthMint(totalSupply(), bTokenFee);
```

Comment:
Don't we need to take into account the marketSpread here?

Reply:
Yes, we do.